### DISCUSSION TITLE
Export data about laser points automatically detected

### DISCUSSION THREAD

**Discussion by BastienTaormina** [biigle/core#537] [Q&A]
Hello,

I am trying to use Biigle in order to measure the distance between two laser points on ROV recordings.
I am then using the automatic point detection on a lot of images (several thousand in each volume). The algorithm is working pretty well, except a few error.

I would like then to export the data obtained, i.e. for each of my picture in my volume , having the number of pixel calculated between my two laser points. Unfortunately, as the detected laser points are only here to calibrate the picture in Biigle (that's what I understood), these are not considered as "label". So when I am trying to export the data, my file is empty (or almost empty, there is only the points I manually annotated to help the automatic detection). Considering the high number of pictures I am working on, I cannot create manually an object in each picture of the volume. 

Thus, I was wandering if there is any method to export the data obtained about the laser points detected?

Don't hesitate if any additional information are needed.

**Comment by mzur** [comment]
You are right, there is no report that will provide you with all detected laser points if there are no annotations on the image. However, you can fetch the points via the [API](https://biigle.de/doc/api/index.html#api-Images-ImagesShowLaserpoints). Example: <https://biigle.de/api/v1/images/2746744/laserpoints>

Take a look at the [community scripts](https://github.com/biigle/community-scripts) for Python script examples that use the API.

**Reply by BastienTaormina** [reply]
Thanks Martin.
Unfortunately, I am not a Python user, so it's going to be complicated for me to use this kind of method.

**Reply by mzur** [reply]
I understand. Unfortunately I don't have the bandwidth to write a script for you right now, sorry. Maybe you can ask someone else.

**Reply by dlangenk** [reply]
If you are familiar with any programming language you can also use www.biigle.de/api/v1/volumes/:id/images/filter/laserpoints (replace :id with the volume ID) and then for each of the numbers you receive you can call the url @mzur provided: https://biigle.de/api/v1/images/:id/laserpoints (replace :id with the numbers from the previous call).

**Reply by BastienTaormina** [reply]
Thanks a lot for the feedback Daniel.
I am using R, and I am right now trying to use it to extract the data with API, but everything is quite new for me  😄 
Your input should help me to find a way

**Reply by BastienTaormina** [reply]
I am having troubles to access API from R, I thinks it is because I don't know where to find and to add the authentication token...
This is the line I am using to try to access the ID data of my volume:
`res = GET("https://biigle.de/api/v1/volumes/8845/images/filter/laserpoints")`
Do you have any ideas?

**Reply by BastienTaormina** [reply]
I found a way :)

**Reply by mzur** [reply]
Great, an R script is totally fine, too. If you like, you could share the final script in the [community scripts](https://github.com/biigle/community-scripts).

**Reply by mzur** [reply]
I'm sorry but I just remembered that there is a [convert laser points](https://github.com/biigle/community-scripts/tree/master/convert-laser-points) community script. This will transform automatically detected laser points into point annotations so they will be included in reports! It's a Python script, though.

**Reply by BastienTaormina** [reply]
Don't worry, I saw that script during my research but considering that it was Python, I didn't understand. I will try to clean and share my R script for people more familiar with this language!
